artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionImpl.java

```diff
@@ -298,6 +299,28 @@ public void createAddress(final SimpleString address, RoutingType routingType, b
       createAddress(address, EnumSet.of(routingType), autoCreated);
    }
 
+   @Override
+   public void createQueue(QueueConfiguration queueConfiguration) throws ActiveMQException {
+      internalCreateQueue(queueConfiguration);
+   }
+
+   @Override
+   public void createSharedQueue(QueueConfiguration queueConfiguration) throws ActiveMQException {
+      checkClosed();
+
+      startCall();
+      try {
+         sessionContext.createSharedQueue(queueConfiguration);
+      } finally {
+         endCall();
+      }
+   }
+
+   @Override
+   public void createTemporaryQueue(QueueConfiguration queueConfiguration) throws ActiveMQException {
```

Comment:
I'm wondering, do we still need all these overloads like `createSharedQueue` and `createTemporaryQueue`. Shouldn't the intent be expressed by the state of `QueueConfiguration` object itself? As the client of the api, I wouldn't expect that the configuration I am passing will be mutated.

Reply:
I would agree, essentially a shared queue or temporary queue is just all attributes of a queue. As such having a more simplified single createQueue method.

Reply:
I'm in favor of that. I'll take those out to simplify it even further.